graph/path/yen_ksp.go

```diff
@@ -123,7 +130,13 @@ func (g yenKSPAdjuster) Edge(uid, vid int64) graph.Edge {
 }
 
 func (g yenKSPAdjuster) Weight(xid, yid int64) (w float64, ok bool) {
-	return g.g.Weight(xid, yid)
+	var weight Weighting
```

Comment:
Don't do this each time. Add a `weight func(xid, yid int64) (w float64, of bool)` field that you determined in the preamble for `YenKShortest` and call that here as

```
func (g yenKSPAdjuster) Weight(xid, yid int64) (w float64, ok bool) {
    return g.weight(xid, yid)
}
```